### PR TITLE
CLOUDP-82587: database users controller waits for clusters

### PR DESCRIPTION
**Pull request by antonlisovenko**
Note, that this PR needs to be reviewed after https://github.com/mongodb/mongodb-atlas-kubernetes/pull/121

This PR adds the code to support waiting for clusters before marking database users as ready.

Note, that this is not fully working now as needs the support for https://docs.atlas.mongodb.com/reference/api/clusters-check-operation-status/ done by mongodb-atlas (https://jira.mongodb.org/browse/CLOUDP-83026). Though the code to read clusters and wait for them is added.

The tests are extended to:
* create real clusters
* check that db users were created in atlas
* wait for clusters to get updated
* check the db connectivity and roles

### CODE REVIEW

File: pkg/controller/atlasdatabaseuser/databaseuser.go [hunk 1]
```diff
@@ -26,16 +33,109 @@ func (r *AtlasDatabaseUserReconciler) ensureDatabaseUser(ctx *workflow.Context,
 				return workflow.Terminate(workflow.DatabaseUserNotCreatedInAtlas, err.Error())
 			}
 			ctx.Log.Infow("Created Atlas Database User", "name", dbUser.Spec.Username)
-			return workflow.OK()
+			return retryAfterUpdate
```
Comment (antonlisovenko) [inline pkg/controller/atlasdatabaseuser/databaseuser.go:36]: now we requeue the reconciliation after each create/update

File: pkg/controller/atlasdatabaseuser/databaseuser.go [hunk 2]
```diff
@@ -26,16 +33,109 @@ func (r *AtlasDatabaseUserReconciler) ensureDatabaseUser(ctx *workflow.Context,
...
 }
+
+func cluserIsReady(client mongodbatlas.Client, projectID, clusterName string) (bool, error) {
+	// TODO CLOUDP-83026 GET for clusters isn't working, we need the https://docs.atlas.mongodb.com/reference/api/clusters-check-operation-status/
+	_ = fmt.Sprintf("%v %v %v", client, projectID, clusterName)
+	if projectID == "xyz" {
+		// Cheating the linter
+		return false, errors.New("boo")
+	}
+	return true, nil
```
Comment (antonlisovenko) [inline pkg/controller/atlasdatabaseuser/databaseuser.go:99]: so far we always return `true` until the atlas-client is updated

File: test/int/dbuser_test.go [hunk 3]
```diff
@@ -39,49 +49,120 @@ var _ = Describe("AtlasDatabaseUser", func() {
...
-		// So far we don't need the cluster
-		// By("Creating cluster", func() {
-		//	createdCluster = testAtlasCluster(namespace.Name, "test-cluster", createdProject.Name)
-		//	Expect(k8sClient.Create(context.Background(), createdCluster)).ToNot(HaveOccurred())
-		//
-		//	Eventually(testutil.WaitFor(k8sClient, createdCluster, status.TrueCondition(status.ReadyType), validateClusterCreatingFunc()),
-		//		1800, interval).Should(BeTrue())
-		// })
+		By("Creating cluster", func() {
+			createdClusterAWS = mdbv1.DefaultAWSCluster(namespace.Name, createdProject.Name)
```
Comment (antonlisovenko) [inline test/int/dbuser_test.go:65]: 2 clusters are created

File: test/int/dbuser_test.go [hunk 4]
```diff
@@ -3,29 +3,39 @@ package int
...
+
 	mdbv1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/project"
 	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
 	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/util/kube"
 	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/util/testutil"
 )
 
 const UserPasswordSecret = "user-password-secret"
+const DevMode = false
```
Comment (antonlisovenko) [inline test/int/dbuser_test.go:27]: this flag allows to develop faster and reuse the project and the cluster. 
Set to `true` when debugging locally

File: test/int/dbuser_test.go [hunk 5]
```diff
@@ -39,49 +49,120 @@ var _ = Describe("AtlasDatabaseUser", func() {
 		Expect(k8sClient.Create(context.Background(), &passwordSecret)).To(Succeed())
 
 		By("Creating the project", func() {
-			createdProject = mdbv1.DefaultProject(namespace.Name, connectionSecret.Name)
+			createdProject = mdbv1.DefaultProject(namespace.Name, connectionSecret.Name).
+				WithIPAccessList(project.NewIPAccessList().WithIP("0.0.0.0/0"))
```
Comment (antonlisovenko) [inline test/int/dbuser_test.go:53]: note, that for this test to pass we need to set the IP Access Lists (to "allow everyone") to check mongodb connectivity

**Review by vasilevp — APPROVED** [review]

**Pull request merged**